File: clients/algoliasearch-client-php/lib/Model/Search/AlternativesAsExact.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * AlternativesAsExact Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class AlternativesAsExact
{
    /**
     * Possible values of this enum
     */
    const IGNORE_PLURALS = 'ignorePlurals';

    const SINGLE_WORD_SYNONYM = 'singleWordSynonym';

    const MULTI_WORDS_SYNONYM = 'multiWordsSynonym';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::IGNORE_PLURALS,
            self::SINGLE_WORD_SYNONYM,
            self::MULTI_WORDS_SYNONYM,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/recommend/AlternativesAsExact.java
package com.algolia.model.recommend;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets alternativesAsExact */
@JsonAdapter(AlternativesAsExact.Adapter.class)
public enum AlternativesAsExact {
  IGNORE_PLURALS("ignorePlurals"),

  SINGLE_WORD_SYNONYM("singleWordSynonym"),

  MULTI_WORDS_SYNONYM("multiWordsSynonym");

  private final String value;

  AlternativesAsExact(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static AlternativesAsExact fromValue(String value) {
    for (AlternativesAsExact b : AlternativesAsExact.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<AlternativesAsExact> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final AlternativesAsExact enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public AlternativesAsExact read(final JsonReader jsonReader)
      throws IOException {
      String value = jsonReader.nextString();
      return AlternativesAsExact.fromValue(value);
    }
  }
}
